### PR TITLE
use asyncio instead of threads for activity task executions

### PR DESCRIPTION
**Pull request by balintzs**
set limit on maximum concurrent executions to be 100;
remove scheduler for thread-count metric logging;

### CODE REVIEW

File: slingshot/core/runners/base.py [hunk 1]
```diff
@@ -17,73 +19,33 @@ class BaseWorkflowRunner(object):
     def __init__(self, auth_config=None, pull_image=False, client=None):
         self.log = logging.getLogger(type(self).__name__)
         self.client = client
-        self._threads_decider = {}
-        self._threads_activity = {}
-        self._threads_status = {}
+
+        self._running = False
+        self._d_thread = threading.Thread(target=self._poll_for_decider_task)
```
Comment (skygeo) [inline slingshot/core/runners/base.py]: Maybe being explicit with the names, and call the things _decider_thread and _decider_queue

File: slingshot/core/runners/base.py [hunk 2]
```diff
@@ -145,12 +94,41 @@ def _poll_for_decider_task(self, thread_id):
...
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        loop.run_until_complete(self._schedule_steps())
+
+    async def _schedule_steps(self):
+        while self._running:
+            self.log.debug("Waiting for activity task...")
+            # Async sleep while the q is empty, then get the next item
+            while self._running and self._a_queue.empty():
+                await asyncio.sleep(1)
```
Comment (skygeo) [inline slingshot/core/runners/base.py]: Hum. You might want to swap with an async queue instead
https://docs.python.org/3/library/asyncio-queue.html

Comment (balintzs) [inline slingshot/core/runners/base.py]: nope... that doesn't work between threads (or at least I couldn't get it to work)

File: slingshot/core/runners/base.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 from slingshot.core.metrics import metric
 from datetime import datetime
+import os
```
Comment (skygeo) [inline slingshot/core/runners/base.py]: remove os

**Comment by skygeo** [comment]
LGTM

**Pull request merged**